src/sentry/sentry_metrics/querying/data_v2/units.py

```diff
@@ -57,9 +57,13 @@ class Unit:
     scaling_factor: float | int
 
     def convert(self, value: float | int) -> float | int:
-        return value * (self.scaling_factor or 1)
+        return value * self.scaling_factor
+
+    def apply_on_timeseries(self, timeseries: Timeseries) -> Timeseries | Formula:
+        # In case the factor is the identity of the multiplication, we do not apply any formula.
+        if self.scaling_factor in {1.0, 1}:
```

Comment:
I clearly specified identity values of both data types instead of doing an int conversion, to avoid depending on external logic.